stream-chat-android-client/src/test/java/io/getstream/chat/android/client/uploader/StreamFileUploaderTest.kt

```diff
@@ -0,0 +1,212 @@
+        )
+        streamFileUploader.sendFile(
+            channelType,
+            channelId,
+            userId,
+            connectionId,
+            File(""),
+            progressCallback
+        )
+
+        verify(retrofitCdnApi).sendFile(
+            eq(channelType),
+            eq(channelId),
+            file = any(),
+            eq(apiKey),
+            eq(userId),
+            eq(connectionId)
+        )
+    }
+
+    @Test
+    fun `Should send image to api when sending image without progress callback`() {
+        whenever(retrofitCdnApi.sendImage(any(), any(), any(), any(), any(), any())).thenReturn(
+            RetroSuccess(UploadFileResponse("file")).toRetrofitCall()
+        )
+        streamFileUploader.sendImage(channelType, channelId, userId, connectionId, File(""))
+
+        verify(retrofitCdnApi).sendImage(
+            eq(channelType),
+            eq(channelId),
+            file = any(),
+            eq(apiKey),
+            eq(userId),
+            eq(connectionId)
+        )
+    }
+
+    @Test
+    fun `Should return file when successfully sent image without progress callback`() {
+        val file = "file"
+        whenever(retrofitCdnApi.sendImage(any(), any(), any(), any(), any(), any())).thenReturn(
+            RetroSuccess(UploadFileResponse(file)).toRetrofitCall()
+        )
+        val result =
+            streamFileUploader.sendImage(channelType, channelId, userId, connectionId, File(""))
+
+        result shouldBeEqualTo file
+    }
+
+    @Test
+    fun `Should return null when sending image without progress callback failed`() {
+        whenever(retrofitCdnApi.sendImage(any(), any(), any(), any(), any(), any())).thenReturn(
+            RetroError<UploadFileResponse>(500).toRetrofitCall()
+        )
+        val result =
+            streamFileUploader.sendImage(channelType, channelId, userId, connectionId, File(""))
+
+        result.shouldBeNull()
+    }
+
+    @Test
+    fun `Should send image to api when sending image with progress callback`() {
+        whenever(retrofitCdnApi.sendImage(any(), any(), any(), any(), any(), any())).thenReturn(
+            RetroSuccess(UploadFileResponse("file")).toRetrofitCall()
+        )
+        streamFileUploader.sendImage(
+            channelType,
+            channelId,
+            userId,
+            connectionId,
+            File(""),
+            progressCallback
+        )
+
+        verify(retrofitCdnApi).sendImage(
+            eq(channelType),
+            eq(channelId),
+            file = any(),
+            eq(apiKey),
+            eq(userId),
+            eq(connectionId)
+        )
+    }
+
+    @Test
+    fun `Should call api delete file when deleting file`() {
+        whenever(retrofitCdnApi.deleteFile(any(), any(), any(), any(), any())).thenReturn(
+            RetroSuccess(CompletableResponse()).toRetrofitCall()
+        )
+        streamFileUploader.deleteFile(channelType, channelId, userId, connectionId, url)
```

Comment:
Nit: There is a classic way of how to structure tests. And usually arrange, act, and assert are divided by empty lines. So for reading purpose it's better to add an empty line before `streamFileUploader.deleteFile(channelType, channelId, userId, connectionId, url)`

Reply:
nice catch, thanks!